packages/react-server/src/test/utilities.ts

```diff
@@ -1,6 +1,37 @@
+import {Server} from 'http';
 import {Context} from 'koa';
+import getPort from 'get-port';
+
 import {KoaNextFunction} from '../types';
 
+export class TestRack {
+  private servers: Server[] = [];
+
+  unmountAll() {
+    this.servers.forEach(server => server.close());
+  }
+
+  async mount(mountFunction: ({port: number, ip: string}) => Server) {
+    const port = await getPort();
+    const ip = 'http://localhost';
+    const server = mountFunction({port, ip});
+
+    this.servers.push(server);
+
+    return {
+      request: async () => {
+        const text = await fetch(`${ip}:${port}`).then(response =>
```

Comment:
the one thing here is I'd maybe just return the whole response since I think it's unlikely all tests universally care about the text.